remote-storage-managers/s3/src/main/java/org/apache/kafka/rsm/s3/S3RemoteLogSegmentContext.java

```diff
@@ -24,59 +24,52 @@
+ *     <li>file name base offset (e.g. {@code 00000000000000000123}).</li>
+ * </ul>
+ */
 public class S3RemoteLogSegmentContext implements RemoteLogSegmentContext {
 
-    public static final String LOG_FILE_NAME_KEY_NAME = "log_file_name";
-    public static final String OFFSET_INDEX_FILE_NAME_KEY_NAME = "offset_index_file_name";
-    public static final String TIME_INDEX_FILE_NAME_KEY_NAME = "time_index_file_name";
-    public static final Schema SCHEMA = new Schema(
-        new Field(LOG_FILE_NAME_KEY_NAME, Type.STRING),
-        new Field(OFFSET_INDEX_FILE_NAME_KEY_NAME, Type.STRING),
-        new Field(TIME_INDEX_FILE_NAME_KEY_NAME, Type.STRING)
-    );
-
-    private final String logFileName;
-    private final String offsetIndexFileName;
-    private final String timeIndexFileName;
+    private static final short VERSION = 0;
 
-    public S3RemoteLogSegmentContext(final String logFileName,
-                                     final String offsetIndexFileName,
-                                     final String timeIndexFileName) {
-        this.logFileName = logFileName;
-        this.offsetIndexFileName = offsetIndexFileName;
-        this.timeIndexFileName = timeIndexFileName;
-    }
-
-    public String logFileName() {
-        return logFileName;
-    }
+    public static final String VERSION_KEY_NAME = "version";
+    public static final String BASE_OFFSET_KEY_NAME = "base_offset";
+    public static final Schema SCHEMA_V0 = new Schema(
+        new Field(VERSION_KEY_NAME, Type.INT16),
+        new Field(BASE_OFFSET_KEY_NAME, Type.INT64)
+    );
+    private final long baseOffset;
 
-    public String offsetIndexFileName() {
-        return offsetIndexFileName;
+    public S3RemoteLogSegmentContext(final long baseOffset) {
+        this.baseOffset = baseOffset;
     }
 
-    public String timeIndexFileName() {
-        return timeIndexFileName;
+    public long baseOffset() {
+        return baseOffset;
     }
 
     @Override
     public byte[] asBytes() {
-        final Struct struct = new Struct(SCHEMA);
-        struct.set(LOG_FILE_NAME_KEY_NAME, logFileName);
-        struct.set(LOG_FILE_NAME_KEY_NAME, logFileName);
-        struct.set(OFFSET_INDEX_FILE_NAME_KEY_NAME, offsetIndexFileName);
-        struct.set(TIME_INDEX_FILE_NAME_KEY_NAME, timeIndexFileName);
-        final ByteBuffer buf = ByteBuffer.allocate(SCHEMA.sizeOf(struct));
+        final Struct struct = new Struct(SCHEMA_V0);
+        struct.set(VERSION_KEY_NAME, VERSION);
+        struct.set(BASE_OFFSET_KEY_NAME, baseOffset);
+        final ByteBuffer buf = ByteBuffer.allocate(SCHEMA_V0.sizeOf(struct));
         struct.writeTo(buf);
         return buf.array();
     }
 
     public static S3RemoteLogSegmentContext fromBytes(final byte[] bytes) {
-        final Struct struct = SCHEMA.read(ByteBuffer.wrap(bytes));
+        // When there are more schema versions, read conditionally.
+        final Struct struct = SCHEMA_V0.read(ByteBuffer.wrap(bytes));
```

Comment:
I prefer version outside of the payload schema. 
- read version, which can be short 
- get the schema for that version and pass the remaining bytes to it build the required structure.

Reply:
Yeah, I though this to be done when version 1 appear as it should be compatible on the byte level. But you're right, better to do this way from the beginning. Done.